provider/src/main/kotlin/org/gradle/kotlin/dsl/KotlinBuildScript.kt

```diff
@@ -34,7 +35,13 @@ import kotlin.script.templates.ScriptTemplateDefinition
     scriptFilePattern = ".*\\.gradle\\.kts")
 @SamWithReceiverAnnotations("org.gradle.api.HasImplicitReceiver")
 @GradleDsl
-abstract class KotlinBuildScript(project: Project) : Project by project {
+abstract class KotlinBuildScript(
+    project: Project,
+    private val scriptHandler: ScriptHandler) : Project by project {
```

Comment:
:x: I would have added it to the `KotlinScriptHost` and passed the `host` to all script templates instead. This would pave the way for uniform `ScriptApi` behaviour without the need for additional arguments.